src/Microsoft.Health.Fhir.CosmosDb/Features/Operations/Export/CosmosExportOrchestratorJob.cs

```diff
@@ -19,24 +22,77 @@ namespace Microsoft.Health.Fhir.CosmosDb.Features.Operations.Export
     [JobTypeId((int)JobType.ExportOrchestrator)]
     public class CosmosExportOrchestratorJob : IJob
     {
+        private const int DefaultNumberOfTimeRangesPerJob = 100;
+
         private readonly IQueueClient _queueClient;
+        private ISearchService _searchService;
 
-        public CosmosExportOrchestratorJob(IQueueClient queueClient)
+        public CosmosExportOrchestratorJob(
+            IQueueClient queueClient,
+            ISearchService searchService)
         {
             _queueClient = EnsureArg.IsNotNull(queueClient, nameof(queueClient));
+            _searchService = EnsureArg.IsNotNull(searchService, nameof(searchService));
         }
 
+        internal int NumberOfTimeRangesPerJob { get; set; } = DefaultNumberOfTimeRangesPerJob;
+
         public async Task<string> ExecuteAsync(JobInfo jobInfo, IProgress<string> progress, CancellationToken cancellationToken)
         {
             EnsureArg.IsNotNull(jobInfo, nameof(jobInfo));
             EnsureArg.IsNotNull(progress, nameof(progress));
 
             var record = JsonConvert.DeserializeObject<ExportJobRecord>(jobInfo.Definition);
             record.QueuedTime = jobInfo.CreateDate; // get record of truth
-
+            var timeRangeSize = (int)record.MaximumNumberOfResourcesPerQuery;
             var groupJobs = await _queueClient.GetJobByGroupIdAsync((byte)QueueType.Export, jobInfo.GroupId, true, cancellationToken);
 
-            if (groupJobs.Count == 1)
+            // for parallel case we enqueue in batches, so we should handle not completed registration
+            if (record.ExportType == ExportJobType.All && record.IsParallel && (record.Filters == null || record.Filters.Count == 0))
+            {
+                var atLeastOneWorkerJobRegistered = false;
+
+                var resourceTypes = string.IsNullOrEmpty(record.ResourceType)
+                                  ? (await _searchService.GetUsedResourceTypes(cancellationToken))
+                                  : record.ResourceType.Split(',');
+                resourceTypes = resourceTypes.OrderByDescending(x => string.Equals(x, "Observation", StringComparison.OrdinalIgnoreCase)).ToList(); // true first, so observation is processed as soon as
+
+                var since = record.Since == null ? new PartialDateTime(DateTime.UnixEpoch).ToDateTimeOffset() : record.Since.ToDateTimeOffset();
+
+                var till = record.Till.ToDateTimeOffset().AddTicks(-1); // -1 is so _till value can be used as _since in the next time based export
+
+                await Parallel.ForEachAsync(resourceTypes, new ParallelOptions { MaxDegreeOfParallelism = 1, CancellationToken = cancellationToken }, async (type, cancel) =>
```

Comment:
Because this isn't attempting to dedupe processing jobs in the event that the orchestrator job gets interrupted and restarts there is a chance that there will be duplicates in the exported data. 
Two processing jobs could be created that will export the same data, which will produce duplicate files in the storage account.